### PR TITLE
Update deprecation notes with example and guide links

### PR DESCRIPTION
**Pull request by daneshk**
## Purpose
$subject

## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests
- [ ] Updated the spec
- [ ] Checked native-image compatibility

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,7 +8,8 @@ Ballerina Regex Library [DEPRECATED]
...
   [![Github issues](https://img.shields.io/github/issues/ballerina-platform/ballerina-standard-library/module/regex.svg?label=Open%20Issues)](https://github.com/ballerina-platform/ballerina-standard-library/labels/module%2Fregex)
 
-**Deprecation Notice:** This library is deprecated and will no longer be maintained or updated. We recommend that you migrate to [ballerina/lang.regexp](https://lib.ballerina.io/ballerina/lang.regexp/latest) library for continued support and updates.
+**Deprecation Notice:** This library is deprecated and will no longer be maintained or updated. Instead, it is recommended to use the [ballerina/lang.regexp](https://lib.ballerina.io/ballerina/lang.regexp/latest) library for continued support and updates. You can find the new regexp types example [here](https://ballerina.io/by-example/regexp-type), the new regexp operations example [here](https://ballerina.io/by-example/regexp-operations) and the regexp feature guide [here](https://ballerina.io/learn/distinctive-language-features/advanced-general-purpose-language-features/#regular-expressions
```
Comment (praneesha) [inline README.md:11]:
```suggestion
**Deprecation Notice:** This library is deprecated and will no longer be maintained or updated. Instead, it is recommended to use the [`ballerina/lang.regexp`](https://lib.ballerina.io/ballerina/lang.regexp/latest) library for continued support and updates. For more information, see the new [regexp types example](https://ballerina.io/by-example/regexp-type), [regexp operations example](https://ballerina.io/by-example/regexp-operations), and [regexp feature guide](https://ballerina.io/learn/distinctive-language-features/advanced-general-purpose-language-features/#regular-expressions).
```
We should not be adding hyperlinks with the word `here1`. Instead, the best practice is to have the page/BBE title itself as the hyperlink.

File: ballerina/Module.md [hunk 2]
```diff
@@ -1,5 +1,5 @@
-**Deprecation Notice:** This library is deprecated and will no longer be maintained or updated. We recommend that you migrate to [ballerina/lang.regexp](https://lib.ballerina.io/ballerina/lang.regexp/latest) library for continued support and updates.
-
+**Deprecation Notice:** This library is deprecated and will no longer be maintained or updated. Instead, it is recommended to use the [ballerina/lang.regexp](https://lib.ballerina.io/ballerina/lang.regexp/latest) library for continued support and updates. You can find the new regexp types example [here](https://ballerina.io/by-example/regexp-type), the new regexp operations example [here](https://ballerina.io/by-example/regexp-operations) and the regexp feature guide [here](https://ballerina.io/learn/distinctive-language-features/advanced-general-purpose-language-features/#regular-expressions
```
Comment (praneesha) [inline ballerina/Module.md:1]:
```suggestion
**Deprecation Notice:** This library is deprecated and will no longer be maintained or updated. Instead, it is recommended to use the [ballerina/lang.regexp](https://lib.ballerina.io/ballerina/lang.regexp/latest) library for continued support and updates. You can find the new regexp types example [here](https://ballerina.io/by-example/regexp-type), the new regexp operations example [here](https://ballerina.io/by-example/regexp-operations) and the regexp feature guide [here](https://ballerina.io/learn/distinctive-language-features/advanced-general-purpose-language-features/#regular-expressions
```
Same as above.

File: ballerina/Package.md [hunk 3]
```diff
@@ -1,5 +1,5 @@
-**Deprecation Notice:** This library is deprecated and will no longer be maintained or updated. We recommend that you migrate to [ballerina/lang.regexp](https://lib.ballerina.io/ballerina/lang.regexp/latest) library for continued support and updates.
-
+**Deprecation Notice:** This library is deprecated and will no longer be maintained or updated. Instead, it is recommended to use the [ballerina/lang.regexp](https://lib.ballerina.io/ballerina/lang.regexp/latest) library for continued support and updates. You can find the new regexp types example [here](https://ballerina.io/by-example/regexp-type), the new regexp operations example [here](https://ballerina.io/by-example/regexp-operations) and the regexp feature guide [here](https://ballerina.io/learn/distinctive-language-features/advanced-general-purpose-language-features/#regular-expressions
```
Comment (praneesha) [inline ballerina/Package.md:1]:
```suggestion
**Deprecation Notice:** This library is deprecated and will no longer be maintained or updated. Instead, it is recommended to use the [ballerina/lang.regexp](https://lib.ballerina.io/ballerina/lang.regexp/latest) library for continued support and updates. You can find the new regexp types example [here](https://ballerina.io/by-example/regexp-type), the new regexp operations example [here](https://ballerina.io/by-example/regexp-operations) and the regexp feature guide [here](https://ballerina.io/learn/distinctive-language-features/advanced-general-purpose-language-features/#regular-expressions
```
Same as above.

File: README.md (hunk 1 above)
Comment (daneshk) [reply README.md]: Fixed in d934dae

File: ballerina/Module.md (hunk 2 above)
Comment (daneshk) [reply ballerina/Module.md]: Fixed in d934dae

File: ballerina/Package.md (hunk 3 above)
Comment (daneshk) [reply ballerina/Package.md]: Fixed in d934dae

File: README.md [hunk 4]
```diff
@@ -8,7 +8,7 @@ Ballerina Regex Library [DEPRECATED]
...
   [![Github issues](https://img.shields.io/github/issues/ballerina-platform/ballerina-standard-library/module/regex.svg?label=Open%20Issues)](https://github.com/ballerina-platform/ballerina-standard-library/labels/module%2Fregex)
 
-**Deprecation Notice:** This library is deprecated and will no longer be maintained or updated. We recommend that you migrate to [ballerina/lang.regexp](https://lib.ballerina.io/ballerina/lang.regexp/latest) library for continued support and updates.
+**Deprecation Notice:** This library is deprecated and will no longer be maintained or updated. Instead, it is recommended to use the [`ballerina/lang.regexp`](https://lib.ballerina.io/ballerina/lang.regexp/latest) library for continued support and updates. For more information, see the new [regexp types example](https://ballerina.io/by-example/regexp-type), [regexp operations example](https://ballerina.io/by-example/regexp-operations), and [regexp feature guide](https://ballerina.io/learn/distinctive-language-features/advanced-general-purpose-language-features/#regular-expressions).
```
Comment (LakshanWeerasinghe) [inline README.md:11]:
```suggestion
**Deprecation Notice:** This library is deprecated and will no longer be maintained or updated. Instead, it is recommended to use the [`ballerina/lang.regexp`](https://lib.ballerina.io/ballerina/lang.regexp/latest) library for continued support and updates. For more information, see the new [RegExp type example](https://ballerina.io/by-example/regexp-type), [RegExp operations example](https://ballerina.io/by-example/regexp-operations), and [Regular expressions feature guide](https://ballerina.io/learn/distinctive-language-features/advanced-general-purpose-language-features/#regular-expressions).
```

Comment (LakshanWeerasinghe) [reply README.md:11]: @pcnfernando

**Review by praneesha — APPROVED** [review]

File: README.md (hunk 4 above)
Comment (pcnfernando) [reply README.md:11]: Agree with @LakshanWeerasinghe suggestion

**Comment by LakshanWeerasinghe** [comment]
@daneshk changes need to  be applied to other two files as well.

File: ballerina/Module.md [hunk 5]
```diff
@@ -1,4 +1,4 @@
-**Deprecation Notice:** This library is deprecated and will no longer be maintained or updated. We recommend that you migrate to [ballerina/lang.regexp](https://lib.ballerina.io/ballerina/lang.regexp/latest) library for continued support and updates.
+**Deprecation Notice:** This library is deprecated and will no longer be maintained or updated. Instead, it is recommended to use the [`ballerina/lang.regexp`](https://lib.ballerina.io/ballerina/lang.regexp/latest) library for continued support and updates. For more information, see the new [regexp types example](https://ballerina.io/by-example/regexp-type), [regexp operations example](https://ballerina.io/by-example/regexp-operations), and [regexp feature guide](https://ballerina.io/learn/distinctive-language-features/advanced-general-purpose-language-features/#regular-expressions).
```
Comment (daneshk) [inline ballerina/Module.md:1]:
```suggestion
**Deprecation Notice:** This library is deprecated and will no longer be maintained or updated. Instead, it is recommended to use the [`ballerina/lang.regexp`](https://lib.ballerina.io/ballerina/lang.regexp/latest) library for continued support and updates. For more information, see the new [RegExp type example](https://ballerina.io/by-example/regexp-type), [RegExp operations example](https://ballerina.io/by-example/regexp-operations), and [Regular expressions feature guide](https://ballerina.io/learn/distinctive-language-features/advanced-general-purpose-language-features/#regular-expressions).
```

File: ballerina/Package.md [hunk 6]
```diff
@@ -1,5 +1,4 @@
-**Deprecation Notice:** This library is deprecated and will no longer be maintained or updated. We recommend that you migrate to [ballerina/lang.regexp](https://lib.ballerina.io/ballerina/lang.regexp/latest) library for continued support and updates.
-
+**Deprecation Notice:** This library is deprecated and will no longer be maintained or updated. Instead, it is recommended to use the [`ballerina/lang.regexp`](https://lib.ballerina.io/ballerina/lang.regexp/latest) library for continued support and updates. For more information, see the new [regexp types example](https://ballerina.io/by-example/regexp-type), [regexp operations example](https://ballerina.io/by-example/regexp-operations), and [regexp feature guide](https://ballerina.io/learn/distinctive-language-features/advanced-general-purpose-language-features/#regular-expressions).
```
Comment (daneshk) [inline ballerina/Package.md:1]:
```suggestion
**Deprecation Notice:** This library is deprecated and will no longer be maintained or updated. Instead, it is recommended to use the [`ballerina/lang.regexp`](https://lib.ballerina.io/ballerina/lang.regexp/latest) library for continued support and updates. For more information, see the new [RegExp type example](https://ballerina.io/by-example/regexp-type), [RegExp operations example](https://ballerina.io/by-example/regexp-operations), and [Regular expressions feature guide](https://ballerina.io/learn/distinctive-language-features/advanced-general-purpose-language-features/#regular-expressions).
```

**Comment by daneshk** [comment]
> @daneshk changes need to be applied to other two files as well.

Fixed in [21e6815](https://github.com/ballerina-platform/module-ballerina-regex/pull/414/commits/21e6815f1ea7489ca2520d2a4962b4a067d01d9d)

**Pull request merged**